micro-rdk/src/esp32/conn/wifi_error.rs

```diff
@@ -0,0 +1,171 @@
+#![allow(non_snake_case)]
+use esp_idf_svc::sys::*;
+use core::ffi::c_uint;
+
+impl From<u16> for WifiErrReason {
```

Comment:
I think I'd rather get a compilation error if we try to build on a platform where C's `unsigned int` isn't 32 bits. That's a really rare platform.